pkg/prometheus/promcfg.go

```diff
@@ -2604,6 +2604,12 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 	if len(sc.Spec.KubernetesSDConfigs) > 0 {
 		configs := make([][]yaml.MapItem, len(sc.Spec.KubernetesSDConfigs))
 		for i, config := range sc.Spec.KubernetesSDConfigs {
+
+			err := (&config).ValidateSelectorRoles()
```

Comment:
validation should be called from `validateKubernetesSDConfigs` in https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/prometheus/resource_selector.go